### ISSUE TITLE
Access control for berglas-secrets-webhook available?

### ISSUE THREAD

**Issue by sonots** [GoogleCloudPlatform/berglas#38]
I followed https://github.com/GoogleCloudPlatform/berglas/tree/master/examples/kubernetes to setup berglas for my k8s cluster, and it seems working fine.

Now, I am thinking about access control for berglas-secrets-webhook. In my understanding, cloudfunction is public. Do you have any plans to support access control or do you think there is no risk even if it is public or do you already have any access controls?

**Comment by sonots** [comment]
I've resolved by myself. It is yet beta feature, but I could remove public access referring this.
https://cloud.google.com/functions/docs/securing/managing-access

**Comment by sonots** [comment]
I could remove public access, but I could not figure out which permission I should add to allow access from GKE. 

I am not attaching (I should attach) a service account to GKE cluster, so google compute engine default service account should be attached at GKE cluster, but

```
 gcloud beta functions add-iam-policy-binding FUNCTION_NAME \
    --member="google_compute_engine_default_service_account" \
    --role="roles/cloudfunctions.invoker"
```

did not work.

**Comment by sethvargo** [comment]
`roles/cloudfunctions.invoker` is the correct IAM permission, so I'm a little bit confused. Are you sure your GKE cluster is running with the default service account? The service account should be in the format:

`12345678908-compute@developer.gserviceaccount.com`

```
 gcloud beta functions add-iam-policy-binding FUNCTION_NAME \
    --member="12345678908-compute@developer.gserviceaccount.com" \
    --role="roles/cloudfunctions.invoker"
```

Where `12345678908` is the project _number_.

**Comment by sonots** [comment]
Yes, it is what I tried.

**Comment by sethvargo** [comment]
Hi @sonots 

And you're sure your cluster is using the default compute service account and not a custom one? If so, I"m not entirely sure. I just tested this on a new cluster with the default service account and invoker permissions and it's working.

**Comment by sonots** [comment]
I am pretty sure I am using the default one.
Hmm, let me try again.

**Comment by sonots** [comment]
It looked berglas works well for deployments, but does not work well for daemonsets when I changed permissions.

![image](https://user-images.githubusercontent.com/2290461/61265771-cca45700-a7cc-11e9-9864-274ce4b9936a.png)

```
+ gcloud beta functions remove-iam-policy-binding berglas-secrets-webhook --project xxxxxxxxxx --region asia-northeast1 --member=allUsers --role=roles/cloudfunctions.invoker
etag: BwWNxL23a8c=
version: 1
+ gcloud beta functions add-iam-policy-binding berglas-secrets-webhook --project xxxxxxxxxx --region asia-northeast1 --member=serviceAccount:xxxxxxxxxxx-compute@developer.gserviceaccount.com --role=roles/cloudfunctions.invoker
bindings:
- members:
  - serviceAccount:xxxxxxxxxxxxxxx-compute@developer.gserviceaccount.com
  role: roles/cloudfunctions.invoker
etag: BwWNxL3Pw9E=
version: 1
```

```
$ kubectl get pod
datadog-agent-x95jz          1/1     Running   2          119s    172.20.1.4    gke-image-search-dev-cluste-api-pool4-ef1eb5aa-rv9s   <none>           <none>
datadog-agent-x95jz          0/1     CrashLoopBackOff   5          4m33s   172.20.1.4    gke-image-search-dev-cluste-api-pool4-ef1eb5aa-rv9s   <none>           <none>
```

kubectl describe pod shows that `InitContainer` with `copy-berglas-bin` is not added.

```
$ kubectl describe pod datadog-agent-x95jz
[omit]
Controlled By:  DaemonSet/datadog-agent
Containers:
  datadog-agent:
    Container ID:  docker://fdbf2023cb10194b98fcea9fd856ba14604baa0de9ec481340145d020b76c580
    Image:         datadog/agent:latest
[omit]
```

**Comment by sonots** [comment]
If you know how to debug `MutatingWebhookConfiguration`, please let me know.
